src/ChainOfResponssibility/Poker/Validators/HandValidators/StraightFlushValidation.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System.Linq;
+
+namespace ChainOfResponssibility.Poker.Validators.HandValidators
+{
+    public class StraightFlushValidation : HandValidation
+    {
+        private StraightValidation straight;
+        private FlushValidation flush;
+
+        public StraightFlushValidation()
+        {
+            straight = new StraightValidation();
+            flush = new FlushValidation();
+        }
+
+        protected override string IsValid(Card[] hand)
+        {
+            if ((!string.IsNullOrEmpty(flush.Validate(hand))) && (!string.IsNullOrEmpty(straight.Validate(hand))))
```

Comment:
@aoancea changed to use validators of straight and flush.
A tree type validation is not really possible since we want the first to match to give us the message and the rest of the chain is ignored